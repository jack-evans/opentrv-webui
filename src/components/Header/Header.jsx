import React from 'react'

const Header = ({isOpen, onClick}) => {
  const buttonClassName = isOpen ? 'Header__left-nav-toggle Header__left-nav-toggle--active' : 'Header__left-nav-toggle'
  return (
    <header className='Header'>
      <div className='Header__left-nav-toggle-container'>
        <button type='button' className={buttonClassName} onClick={onClick}>
          <span className='Header__left-nav-icon' />
        </button>
      </div>
      <a className='Header__title' href='/'>
        <h3>OpenTRV</h3>
      </a>
    </header>
  )
}

export default Header
