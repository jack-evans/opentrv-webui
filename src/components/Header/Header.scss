$header-background-color: #222;
$header-height: 60px;
$header-padding-top-bottom: 10px;
$header-padding-left-right: 5px;
$header-logo-height-width: $header-height;
$header-title-line-height: 31px;

.Header {

  .Header__top-nav {
    background-color: $header-background-color;
    display: flex;
    height: $header-height;

    .Header__left-container {
      .Header__title {
        color: white;
        line-height: $header-title-line-height;
      }

      .Header__menu {
        background-color: #ccc;

        .Header__menu-button {
          border: none;
          height: $header-logo-height-width;
          width: $header-logo-height-width;
        }

        .Header__menu-icon {
        }
      }
    }

    .Header__right-container {

    }
  }

  .Header__left-nav {
    background-color: #f0f2f4;
    width: 250px;
    height: 100%;
    display: flex;
    position: fixed; /* Stay in place */
    z-index: 1; /* Stay on top */
    top: 60px;
    left: 0;
    overflow-x: hidden; /* Disable horizontal scroll */
    padding-top: 60px; /* Place content 60px from the top */
  }
}