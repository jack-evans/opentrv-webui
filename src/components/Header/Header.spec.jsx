import React from 'react'
import Header from './Header'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('Header..jsx', () => {
  it('renders without crashing', () => {
    const cbFunction = () => {}

    const expectedRender = (
      <header className='Header'>
        <div className='Header__left-nav-toggle-container'>
          <button type='button' className='Header__left-nav-toggle' onClick={cbFunction}>
            <span className='Header__left-nav-icon' />
          </button>
        </div>
        <a className='Header__title' href='/'>
          <h3>Welcome to React</h3>
        </a>
      </header>
    )
    expect(shallow(<Header onClick={cbFunction} />).contains(expectedRender)).toBe(true)
  })

  describe('when the button is clicked', () => {
    it('calls the callback function', () => {
      const cbFunction = () => {
        expect(1).toEqual(1)
      }

      const wrapper = shallow(<Header onClick={cbFunction} />)
      wrapper.find('button').simulate('click')
    })
  })

  describe('when the isOpen property is true', () => {
    it('has applied the Header__left-nav-toggle--active class to the button', () => {
      const cbFunction = () => {}

      const expectedRender = (
        <header className='Header'>
          <div className='Header__left-nav-toggle-container'>
            <button type='button' className='Header__left-nav-toggle Header__left-nav-toggle--active' onClick={cbFunction}>
              <span className='Header__left-nav-icon' />
            </button>
          </div>
          <a className='Header__title' href='/'>
            <h3>Welcome to React</h3>
          </a>
        </header>
      )
      expect(shallow(<Header onClick={cbFunction} isOpen />).contains(expectedRender)).toBe(true)
    })
  })
})
