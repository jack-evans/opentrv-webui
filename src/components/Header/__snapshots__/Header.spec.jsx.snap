// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header..jsx renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="Header__left-nav-toggle-container"
        >
          <button
            className="Header__left-nav-toggle"
            onClick={[Function]}
            type="button"
          >
            <span
              className="Header__left-nav-icon"
            />
          </button>
        </div>,
        <a
          className="Header__title"
          href="/overview"
        >
          <h3>
            OpenTRV
          </h3>
        </a>,
      ],
      "className": "Header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="Header__left-nav-toggle"
            onClick={[Function]}
            type="button"
          >
            <span
              className="Header__left-nav-icon"
            />
          </button>,
          "className": "Header__left-nav-toggle-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="Header__left-nav-icon"
            />,
            "className": "Header__left-nav-toggle",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "Header__left-nav-icon",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": "button",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3>
            OpenTRV
          </h3>,
          "className": "Header__title",
          "href": "/overview",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "OpenTRV",
          },
          "ref": null,
          "rendered": "OpenTRV",
          "type": "h3",
        },
        "type": "a",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="Header__left-nav-toggle-container"
          >
            <button
              className="Header__left-nav-toggle"
              onClick={[Function]}
              type="button"
            >
              <span
                className="Header__left-nav-icon"
              />
            </button>
          </div>,
          <a
            className="Header__title"
            href="/overview"
          >
            <h3>
              OpenTRV
            </h3>
          </a>,
        ],
        "className": "Header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="Header__left-nav-toggle"
              onClick={[Function]}
              type="button"
            >
              <span
                className="Header__left-nav-icon"
              />
            </button>,
            "className": "Header__left-nav-toggle-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="Header__left-nav-icon"
              />,
              "className": "Header__left-nav-toggle",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "Header__left-nav-icon",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            "type": "button",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3>
              OpenTRV
            </h3>,
            "className": "Header__title",
            "href": "/overview",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "OpenTRV",
            },
            "ref": null,
            "rendered": "OpenTRV",
            "type": "h3",
          },
          "type": "a",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Header..jsx when the isOpen property is true has applied the Header__left-nav-toggle--active class to the button 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    isOpen={true}
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="Header__left-nav-toggle-container"
        >
          <button
            className="Header__left-nav-toggle Header__left-nav-toggle--active"
            onClick={[Function]}
            type="button"
          >
            <span
              className="Header__left-nav-icon"
            />
          </button>
        </div>,
        <a
          className="Header__title"
          href="/overview"
        >
          <h3>
            OpenTRV
          </h3>
        </a>,
      ],
      "className": "Header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="Header__left-nav-toggle Header__left-nav-toggle--active"
            onClick={[Function]}
            type="button"
          >
            <span
              className="Header__left-nav-icon"
            />
          </button>,
          "className": "Header__left-nav-toggle-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="Header__left-nav-icon"
            />,
            "className": "Header__left-nav-toggle Header__left-nav-toggle--active",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "Header__left-nav-icon",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": "button",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3>
            OpenTRV
          </h3>,
          "className": "Header__title",
          "href": "/overview",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "OpenTRV",
          },
          "ref": null,
          "rendered": "OpenTRV",
          "type": "h3",
        },
        "type": "a",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="Header__left-nav-toggle-container"
          >
            <button
              className="Header__left-nav-toggle Header__left-nav-toggle--active"
              onClick={[Function]}
              type="button"
            >
              <span
                className="Header__left-nav-icon"
              />
            </button>
          </div>,
          <a
            className="Header__title"
            href="/overview"
          >
            <h3>
              OpenTRV
            </h3>
          </a>,
        ],
        "className": "Header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="Header__left-nav-toggle Header__left-nav-toggle--active"
              onClick={[Function]}
              type="button"
            >
              <span
                className="Header__left-nav-icon"
              />
            </button>,
            "className": "Header__left-nav-toggle-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="Header__left-nav-icon"
              />,
              "className": "Header__left-nav-toggle Header__left-nav-toggle--active",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "Header__left-nav-icon",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            "type": "button",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3>
              OpenTRV
            </h3>,
            "className": "Header__title",
            "href": "/overview",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "OpenTRV",
            },
            "ref": null,
            "rendered": "OpenTRV",
            "type": "h3",
          },
          "type": "a",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
